src/shared/api/dto.ts

```diff
@@ -1,183 +1,136 @@
-  }
-
-  interface Role {
-    id: string
-    permissions: PermissionToken[]
-  }
-
-  interface PermissionToken {
-    name: string
-    params: any
-  }
-
-  interface Account {
-    id: string
-    assets: Asset[]
-    signatories?: PublicKey[]
-    permission_tokens?: PermissionToken[]
-    roles: Role[]
-    signature_check_condition?: any
-    metadata: any
-  }
-
-  interface Domain {
-    id: string
-    accounts: Account[]
-    asset_definitions: AssetDefinition[]
-    logo: null | string
-    metadata: any
-    /**
-     * amount of triggers, always 0 for now
-     */
-    triggers: number
-  }
-
-  interface PublicKey {
-    digest_function: string
-    payload: string
-  }
-
-  interface Status {
-    peers: string
-    blocks: string
-    txs: string
-    uptime: {
-      secs: string
-      nanos: string
-    }
-  }
-
-  export interface BlockShallow {
-    height: number
-    /**
-     * ISO DateTime
-     */
-    timestamp: string
-    block_hash: string
-    /**
-     * Transactions count
-     */
-    transactions: number
-    /**
-     * Rejected transactions count
-     */
-    rejected_transactions: number
-  }
-
-  export interface Block {
-    height: number
-    /**
-     * See {@link BlockShallow.timestamp}
-     */
-    timestamp: string
-    block_hash: string
-    parent_block_hash: string
-    rejected_transactions_merkle_root_hash: string
-    invalidated_blocks_hashes: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedValidTransaction}
-     */
-    transactions: string[]
-    /**
-     * List of serialized {@link @iroha2/data-model#VersionedRejectedTransaction}
-     */
-    rejected_transactions: string[]
-    /**
-     * List of hashes. WIP always empty
-     */
-    view_change_proofs: string[]
-  }
-
-  export interface TransactionDto {
-    /**
-     * WIP zeroed
-     */
-    hash: string
-    block_hash: string
-    // TODO: bloch_height is missing from backend response
-    block_height: number
-    payload: TransactionPayload
-    signatures: Signature[]
-    rejection_reason?: string
-  }
-
-  export interface TransactionPayload {
-    account_id: string
-    instructions: TransactionInstructions
-    /**
-     * ISO timestamp
-     */
-    creation_time: string
-    time_to_live_ms: number | null
-    nonce: number | null
-    metadata: any
-  }
-
-  /**
-   * `Instructions` `string[]` - list of serialized
-   * {@link @iroha2/data-model#Instruction}
-   */
-  export type TransactionInstructions = Tagged<'Instructions', string[]> | Tagged<'Wasm', undefined>;
-
-  export interface Signature {
-    /**
-     * Public key's multihash
-     */
-    public_key: string
-    /**
-     * Hex binary
-     */
-    payload: string
-  }
+import { z } from 'zod';
+
+const paginationSchema = z.object({
+  page: z.number(),
+  per_page: z.number(),
+  // TODO: remove nullable when backend is ready
+  total_pages: z.number().nullable(),
+  total_items: z.number().nullable(),
```

Comment:
These numbers are all not _always_ numbers. In the schema they are `BigInt`s, which when _too_ large become serialized as strings in JSON.

To handle it properly, we have to use `BigInt` instead :<

It might look something like this:

```ts
const toBigInt = z.number().or(z.string()).pipe(z.coerce.bigint());
```

Reply:
I think we don't need to worry about it for now. The max safe integer is 9007199254740991 what means we are too far away to that number. Currently, backend perfectly accepts type coming from frontend and backwards so lets keep it as is.

Reply:
Agreed, not practical yet.